### PR TITLE
Drop support for pinning hg.mozilla.org TLS certificate fingerprint

### PR DESCRIPTION
**Pull request by jcristau**
Gecko is dropping this in
https://bugzilla.mozilla.org/show_bug.cgi?id=1793923, and other taskgraph users shouldn't need it either.

### CODE REVIEW

File: src/taskgraph/run-task/run-task [hunk 1]
```diff
@@ -1101,8 +1048,7 @@ def main(args):
         add_vcs_arguments(parser, repository, name)
 
     parser.add_argument('--fetch-hgfingerprint', action='store_true',
-                        help='Fetch the latest hgfingerprint from the secrets store, '
-                        'using the taskclsuerProxy')
+                        help="Obsolete option, ignored for backwards compatibility")
```
Comment (ahal) [inline src/taskgraph/run-task/run-task:1051]: Might be better to set `help=argparse.SUPPRESS`. That way the option won't even show up in the help in the first place.

File: test/test_transforms_job_run_task.py [hunk 2]
```diff
@@ -47,7 +47,7 @@ def assert_docker_worker(task):
         "extra": {},
         "label": "fake-task-label",
         "routes": [],
-        "scopes": ["secrets:get:project/taskcluster/gecko/hgfingerprint"],
+        "scopes": [],
```
Comment (ahal) [inline test/test_transforms_job_run_task.py:50]: I guess we can't remove the scope from ci-config until the Gecko patch rides the trains everywhere..

**Review by ahal — APPROVED** [review]
Nice, thanks for this!

File: test/test_transforms_job_run_task.py (hunk 2 above)
Comment (jcristau) [reply test/test_transforms_job_run_task.py:50]: Right.  I'm planning to let it ride 108, and uplift to ESR 102.6 or 102.7.

File: src/taskgraph/run-task/run-task (hunk 1 above)
Comment (jcristau) [reply src/taskgraph/run-task/run-task]: Thanks for the tip :)

**Pull request merged**